src/frontends/tensorflow_common/src/op/rank.cpp

```diff
@@ -14,17 +18,35 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 
-ov::OutputVector translate_rank_op(const NodeContext& node) {
-    default_op_checks(node, 1, {"Rank"});
-    auto input = node.get_input(0);
-    auto input_shape = make_shared<v3::ShapeOf>(input, ov::element::i32);
-    auto unsqueeze_input_rank = make_shared<v3::ShapeOf>(input_shape, ov::element::i32);
-    auto input_rank = make_shared<v0::Squeeze>(unsqueeze_input_rank);
+ov::OutputVector translate_rank_op(const NodeContext &node) {
+  default_op_checks(node, 1, {"Rank"}, true);
+  auto input = node.get_input(0);
+  auto complex_type_mark =
+      as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
+  auto input_shape = make_shared<v3::ShapeOf>(input, ov::element::i32);
+  if (complex_type_mark) {
+    input = complex_type_mark->input_value(0);
+
+    auto unsqueeze_input_rank =
+        make_shared<v3::ShapeOf>(input_shape, ov::element::i32);
+    auto input_rank_with_complex =
+        make_shared<v0::Squeeze>(unsqueeze_input_rank);
+    // Squeeze again to eliminate the extra dimension
+    auto input_rank = make_shared<v0::Squeeze>(
+        input_rank_with_complex,
+        make_shared<v0::Constant>(ov::element::i32, Shape{1},
+                                  input_shape->get_input_size() - 1));
```

Comment:
please think how to do. Let say you have complex tensor of shape [3, 4] but since OV does not support complex tensor natively it works with floating-point representation with shape [3, 4, 2], where 2 - is auxiliary. You don't need to count this dimension during rank computation

Reply:
Yes, I misunderstood it before and wanted to "squeeze" the last dimension. I've changed my way to subtract it by 1 and the tests have passed now. Could you please kindly review it again? :)

Reply:
Great job, thank!